engine/enginefuzz_test.go

```diff
@@ -32,9 +33,222 @@ import (
 const testRuns = 100
 
 type testCase struct {
-	query          string
-	loads          []string
-	oldRes, newRes *promql.Result
+	query              string
+	loads              []string
+	oldRes, newRes     *promql.Result
+	oldStats, newStats *stats.Statistics
+	start, end         time.Time
+	interval           time.Duration
+}
+
+// validateExpr checks if the expr should be skipped from fuzz tests.
+func validateExpr(expr parser.Expr) bool {
+	valid := true
+
+	parser.Inspect(expr, func(node parser.Node, path []parser.Node) error {
+		switch n := node.(type) {
+		case *parser.Call:
+			if n.Func.Name == "scalar" {
+				valid = false
+				return errors.New("error")
+			}
+		}
+		return nil
+	})
+	return valid
+}
+
+func FuzzEnginePromQLSmithQueryStatsRangeQuery(f *testing.F) {
```

Comment:
Do we need two new fuzz tests? couldnt we incorporate this into existing fuzz tests?